### ISSUE TITLE
Refactoring Data Catalog

### ISSUE THREAD

**Issue by Amustache** [hestiaAI/data-catalog#5]

**Comment by alexbfree** [comment]
A few thoughts:

- Yes, a standard structure for all such information is a good thing we should be moving towards.
- We can't assume people will be willing to share raw data publicly or without conditions. I think consent will always need to be handled sensitively and we should not be moving towards centralised storage of personal data (apart from anything else this has GDPR implications). 
- If we want to store 'raw' data I think we need to have some process of anonymisation of data before it is stored. Or generation of dummy data to match a given format.
- I think `actors` is a confusing term to use. Actors sounds like end users and individuals. We are talking about data-holding organisations. I think `holder` or `organisation` would be a clearer term. 
- The nesting of models underneath each company doesn't reflect the fact that we want to be able to have models that can support multiple companies. I think we should be careful about pushing ourselves too far down the 'thinking of only one company at a time' route. For example another approach might be to group similar types of data holding organisations / datasets together.

**Comment by Amustache** [comment]
> We can't assume people will be willing to share raw data publicly or without conditions. I think consent will always need to be handled sensitively and we should not be moving towards centralised storage of personal data (apart from anything else this has GDPR implications). So as it stands I don't think we should do the input folder as proposed).

> If we want to store 'raw' data (for the `input` folder) I think we need to have some process of anonymisation of data before it is stored. Or generation of dummy data to match a given format.

Agreed. Firstly, the `input` folder is .gitignored by default; secondly, this is something that we should in fact discuss beforehand. For now, I think this is not a priority, models and scripts are.

> I think `actors` is a confusing term to use. Actors sounds like end users and individuals. We are talking about data-holding organisations. I think `holder` or `organisation` would be a clearer term.

Agreed. Maybe we can agree on `entities`, which is the nomenclature used by [Disconnect](https://github.com/disconnectme) and other such services?

> The nesting of models underneath each company doesn't reflect the fact that we want to be able to have models that can support multiple companies. I think we should be careful about pushing ourselves too far down the 'thinking of only one company at a time' route. For example another approach might be to group similar types of data holding organisations / datasets together.

Yes, we can discuss that again! For now (and it can change), the idea is a _model_ per company, and such model as _semantic descriptions_, that can be reused.

**Comment by Amustache** [comment]
I updated the initial issue with your inputs, @alexbfree - let me know if it is better!

**Comment by alexbfree** [comment]
This is moving in the right direction (and I like the way you have updated this with a changelog)

Further comments:
- I like the use of `entities`, ties in with some other design work i have done in this space where I used that term.
- I think our categories should possibly be semantic. also we should be categorising data, not categorising entities. I think we probably need to allow for both company specific and general semantic types of data. So maybe something like:

```
  entity-specific
     Facebook
        ...
     Twitter
        ...
  data-by-type
     location-data
        ...
     interests-data
        ...
     ads-and-targeting-data
        ...
     activity-data
        ...
     social-history-data
        ...
     data-flow-history
        ...  
```

etc. 

Where `activity-data` would be things like web visit history, app usage history  etc, `social-history-data` could be things like social connections or interactions between people, `data-flow-history` could encompass things like tracker control connections history or records of when apps have shared data, etc..... 

over time we would hope to have more under `data-by-type` than `entity-specific`

**Comment by Amustache** [comment]
Thank you for your review!

> [...] we should be categorising data, not categorising entities [...]

Would it be possible to have more insights about the second point? I'm not sure I understand correctly the data-by-type.

Does that mean (something along) we would have a (complete) data model for Google under "entity/Google", and a (partial, filtered) data model for Google under "data-by-type/location"?

**Comment by Amustache** [comment]
This discussion is tightly related to https://github.com/hestiaAI/internal-discussions/discussions/32, when discussion concepts, relations, etc.

**Comment by alexbfree** [comment]
just discussed with @Amustache - he is going to have an attempt at this refactoring on a branch - so that there can be a pull request to review & potentially merge.
